### ISSUE TITLE
az aks list fails with 'ContainerServiceClientConfiguration' object has no attribute 'api_version'

### ISSUE THREAD

**Issue by Superzer0** [Azure/azure-cli#24274]
## az aks list fails with 'ContainerServiceClientConfiguration' object has no attribute 'api_version'

**Command Name**
`az aks list`

**Errors:**
```
The command failed with an unexpected error. Here is the traceback:
'ContainerServiceClientConfiguration' object has no attribute 'api_version'
Traceback (most recent call last):
  File "/home/linuxbrew/.linuxbrew/Cellar/azure-cli/2.41.0/libexec/lib/python3.10/site-packages/knack/cli.py", line 233, in invoke
    cmd_result = self.invocation.execute(args)
  File "/home/linuxbrew/.linuxbrew/Cellar/azure-cli/2.41.0/libexec/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 663, in execute
    raise ex
  File "/home/linuxbrew/.linuxbrew/Cellar/azure-cli/2.41.0/libexec/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 726, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/home/linuxbrew/.linuxbrew/Cellar/azure-cli/2.41.0/libexec/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 697, in _run_job
    result = cmd_copy(params)
  File "/home/linuxbrew/.linuxbrew/Cellar/azure-cli/2.41.0/libexec/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 333, in __call__
    return self.handler(*args, **kwargs)
  File "/home/linuxbrew/.linuxbrew/Cellar/azure-cli/2.41.0/libexec/lib/python3.10/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler
    return op(**command_args)
  File "/home/linuxbrew/.linuxbrew/Cellar/azure-cli/2.41.0/libexec/lib/python3.10/site-packages/azure/cli/command_modules/acs/custom.py", line 724, in aks_list
    return _remove_nulls(list(managed_clusters))
  File "/home/linuxbrew/.linuxbrew/Cellar/azure-cli/2.41.0/libexec/lib/python3.10/site-packages/azure/core/paging.py", line 129, in __next__
    return next(self._page_iterator)
  File "/home/linuxbrew/.linuxbrew/Cellar/azure-cli/2.41.0/libexec/lib/python3.10/site-packages/azure/core/paging.py", line 76, in __next__
    self._response = self._get_next(self.continuation_token)
  File "/home/linuxbrew/.linuxbrew/Cellar/azure-cli/2.41.0/libexec/lib/python3.10/site-packages/azure/mgmt/containerservice/v2022_07_01/operations/_managed_clusters_operations.py", line 977, in get_next
    request = prepare_request(next_link)
  File "/home/linuxbrew/.linuxbrew/Cellar/azure-cli/2.41.0/libexec/lib/python3.10/site-packages/azure/mgmt/containerservice/v2022_07_01/operations/_managed_clusters_operations.py", line 962, in prepare_request
    _next_request_params["api-version"] = self._config.api_version
AttributeError: 'ContainerServiceClientConfiguration' object has no attribute 'api_version'

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- `az aks list -o table`

## Environment Summary
```
Linux-5.10.102.1-microsoft-standard-WSL2-x86_64-with-glibc2.35, Ubuntu 20.04.5 LTS
Python 3.10.8
Installer: HOMEBREW

azure-cli 2.41.0

Extensions:
azure-devops 0.25.0
interactive 0.4.5

Dependencies:
msal 1.20.0b1
azure-mgmt-resource 21.1.0b1
```

**Comment by yonzhan** [comment]
route to CXP team

**Comment by PramodValavala-MSFT** [comment]
@Superzer0 This has been fixed with this PR - #24270 and should be part of the next release.

In the meantime, the workaround would be to downgrade to 2.40.0

**Comment by Superzer0** [comment]
Ok 👍🏻  thanks for the update and fix
